src/main/java/Bob.java

```diff
@@ -0,0 +1,207 @@
+    private static final String LINE_PREFIX = "    ";
+
+    private static String argument = "";
+    private static final List<Task> list = new ArrayList<>();
+
+    private enum Command {
+        BYE("bye") {
+            @Override
+            public void run() {
+                exit();
+            }
+        },
+        LIST("list") {
+            @Override
+            public void run() {
+                if (list.isEmpty()) {
+                    say("You have not added any tasks yet.");
+                } else {
+                    StringBuilder text = new StringBuilder();
+                    int i;
+                    for (i = 0; i < list.size() - 1; ++i) {
+                        text.append(i + 1).append(".").append(list.get(i)).append("\n");
+                    }
+                    text.append(i + 1).append(".").append(list.get(i));
+                    say(text.toString());
+                }
+            }
+        },
+        TODO("todo") {
+            @Override
+            public void run() {
+                if (argument.isBlank()) {
+                    throw new MissingArgumentException("description of the todo");
+                }
+                Task task = new Todo(argument.strip());
+                list.add(task);
+                say("added: " + task);
+            }
+        },
+        DEADLINE("deadline") {
+            @Override
+            public void run() {
+                int byIndex = argument.lastIndexOf("/by ");
+                if (byIndex == -1) {
+                    throw new MissingArgumentException("'by' argument to add a deadline");
+                }
+                String desc = argument.substring(0, byIndex).strip();
+                if (desc.isBlank()) {
+                    throw new MissingArgumentException("description of the deadline");
+                }
+                Task task = new Deadline(
+                        desc,
+                        argument.substring(byIndex + 4).strip()
+                );
+                list.add(task);
+                say("added: " + task);
+            }
+        },
+        EVENT("event") {
+            @Override
+            public void run() {
+                int fromIndex = argument.lastIndexOf("/from ");
+                int toIndex = argument.lastIndexOf("/to ");
+                if (fromIndex == -1 || toIndex == -1) {
+                    throw new MissingArgumentException("'from' and 'to' arguments to add an event");
+                } else {
+                    Task task = getTask(fromIndex, toIndex);
+                    list.add(task);
+                    say("added: " + task);
+                }
+            }
+
+            private static Task getTask(int fromIndex, int toIndex) {
+                String desc = fromIndex < toIndex
+                                ? argument.substring(0, fromIndex)
+                                : argument.substring(0, toIndex);
```

Comment:
Not sure whether this is accurate? Maybe try to add comments to explain why you choose to incorporate fromIndex and toIndex?